extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftConfig.java

```diff
@@ -183,6 +183,12 @@ public static enum DeploymentResourceKind {
     @ConfigItem
     Optional<List<String>> imagePullSecrets;
 
+    /**
+     * Enable generation of image pull secret, when credentials are available.
+     */
+    @ConfigItem(defaultValue = "true")
```

Comment:
I would disable this property by default, since it should not be used for security reasons.